Mage.Sets/src/mage/cards/y/YarusRoarOfTheOldGods.java

```diff
@@ -0,0 +1,120 @@
+    private static final FilterCreaturePermanent filter =
+            new FilterCreaturePermanent("face-down creature");
+
+    static {
+        filter.add(TargetController.YOU.getControllerPredicate());
+        filter.add(FaceDownPredicate.instance);
+    }
+
+    public YarusRoarOfTheOldGods(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{R}{G}");
+
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.CENTAUR);
+        this.subtype.add(SubType.DRUID);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // Other creatures you control have haste.
+        this.addAbility(new SimpleStaticAbility(new GainAbilityControlledEffect(
+                HasteAbility.getInstance(), Duration.WhileOnBattlefield,
+                StaticFilters.FILTER_PERMANENT_CREATURES, true)));
+        // Whenever one or more face-down creatures you control deal combat damage to a player, draw a card.
+        this.addAbility(new DealCombatDamageControlledTriggeredAbility(
+                Zone.BATTLEFIELD, new DrawCardSourceControllerEffect(1),
+                filter, SetTargetPointer.PLAYER, false)
+                .setTriggerPhrase("Whenever one or more face-down creatures you control deal combat damage to a player, "));
+        // Whenever a face-down creature you control dies, return it to the battlefield face down under its owner's control if it's a permanent card, then turn it face up.
+        this.addAbility(new DiesCreatureTriggeredAbility(
+                new YarusRoarOfTheOldGodsEffect(), false, filter, true));
+    }
+
+    private YarusRoarOfTheOldGods(final YarusRoarOfTheOldGods card) {
+        super(card);
+    }
+
+    @Override
+    public YarusRoarOfTheOldGods copy() {
+        return new YarusRoarOfTheOldGods(this);
+    }
+}
+
+class YarusRoarOfTheOldGodsEffect extends OneShotEffect {
+
+    YarusRoarOfTheOldGodsEffect() {
+        super(Outcome.PutCardInPlay);
+        this.staticText = "return it to the battlefield face down under its owner's control " +
+                "if it's a permanent card, then turn it face up";
+    }
+
+    private YarusRoarOfTheOldGodsEffect(final YarusRoarOfTheOldGodsEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public YarusRoarOfTheOldGodsEffect copy() {
+        return new YarusRoarOfTheOldGodsEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        Card card = game.getCard(getTargetPointer().getFirst(game, source));
+        if (controller == null || card == null || !card.isPermanent(game)) {
+            return false;
+        }
+
+        Ability newSource = source.copy();
+        newSource.setWorksFaceDown(true);
+        MageObjectReference mor = new MageObjectReference(card.getId(),
+                card.getZoneChangeCounter(game), game);
```

Comment:
this feels like the zcc would be off by one but if works in testing then all fine